### PR TITLE
Fixing credentials to use AAD

### PR DESCRIPTION
**Pull request by seankane-msft**
Noticed the `test_*_aad(_async).py` didn't use AAD, instead authenticating with a connection string, this PR fixes that situation.

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration/tests/wrapper.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 from azure.core.exceptions import (
     AzureError,
 )
+from azure.identity import DefaultAzureCredential
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/tests/wrapper.py:15]: Why we need this?

Comment (seankane-msft) [reply sdk/appconfiguration/azure-appconfiguration/tests/wrapper.py:15]: I will remove this

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_aad_async.py [hunk 2]
```diff
@@ -44,9 +46,15 @@ def _delete_setting(self, client, item):
             key=item.key, label=item.label
         )
 
+    def create_aad_client(self, base_url):
+        cred = self.get_credential(AzureAppConfigurationClient, is_async=True)
+        client = AzureAppConfigurationClient(base_url, cred)
+        return AzureAppConfigurationClientProxy(client)
```
Comment (kristapratico) [inline sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_aad_async.py:52]: just curious why this isn't the actual async client

**Review by kristapratico — APPROVED** [review]

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_aad_async.py (hunk 2 above)
Comment (seankane-msft) [reply sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_aad_async.py:52]: Xiang wrote a very clever class to take care of all the async features without having to change the base code. https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/appconfiguration/azure-appconfiguration/tests/async_proxy.py

**Pull request merged**